include/boost/gil/image_processing/scaling.hpp

```diff
@@ -0,0 +1,113 @@
+#include <boost/gil/image_view.hpp>
+#include <boost/gil/rgb.hpp>
+#include <boost/gil/pixel.hpp>
+#include <boost/gil/image_processing/numeric.hpp>
+
+namespace boost{ namespace gil{
+/// \defgroup ScalingAlgorithms
+/// \brief Algorthims suitable for rescaling
+///
+/// These algorithms are used to improve image
+/// quality after image resizing is made.
+
+/// \defgroup DownScalingAlgorithms
+/// \ingroup ScalingAlgorithms
+/// \brief Algorthims suitable for downscaling
+///
+/// These algorithms provide best results when used
+/// for downscaling. Using for upscaling will probably
+/// provide less than good results.
+
+
+/// \brief a single step of lanczos downscaling
+/// \ingroup DownScalingAlgorithms
+///
+/// Use this algorithm to scale down source image
+/// into a smaller image with reasonable quality.
+/// Do note that having a look at the output once
+/// is a good idea, since it might have ringing
+/// artifacts.
+template <typename ImageView>
+void lanczos_at(
+    long int source_x, 
```

Comment:
I noticed lots of trailing whitespaces.

Would it be possible for you to keep them trimmed?
I recommend to enable the `.editorconfig` for your editor, this should take care of it automatically.
See near end of bullet list here https://github.com/boostorg/gil/blob/develop/CONTRIBUTING.md#guidelines